lib/util.js

```diff
@@ -400,11 +400,14 @@ function formatValue(ctx, value, recurseTimes) {
     });
   }
 
+  var constructor = getConstructorOf(value);
+
   // Some type of object without properties can be shortcutted.
   if (keys.length === 0) {
     if (typeof value === 'function') {
-      return ctx.stylize(`[Function${value.name ? `: ${value.name}` : ''}]`,
-                         'special');
+      const ctorName = constructor ? constructor.name : 'Function';
+      return ctx.stylize(
+          `[${ctorName}${value.name ? `: ${value.name}` : ''}]`, 'special');
```

Comment:
Nested templates can be a little bit hard to read, maybe change it to use if-else? (though this LTGM too with this unchanged, now that there is a backport PR already)